cognite/client/_api/simulators/routine_revisions.py

```diff
@@ -0,0 +1,358 @@
+            simulator_external_ids=simulator_external_ids,
+            created_time=created_time,
+        )
+        return self._list_generator(
+            method="POST",
+            limit=limit,
+            url_path="/simulators/routines/revisions/list",
+            resource_cls=SimulatorRoutineRevision,
+            list_cls=SimulatorRoutineRevisionList,
+            chunk_size=chunk_size,
+            filter=filter.dump(),
+            sort=[PropertySort.load(sort).dump()] if sort else None,
+            other_params={"includeAllFields": include_all_fields},
+        )
+
+    @overload
+    def retrieve(self, ids: None = None, external_ids: None = None) -> NoReturn: ...
+
+    @overload
+    def retrieve(self, ids: int, external_ids: None = None) -> SimulatorRoutineRevision | None: ...
+
+    @overload
+    def retrieve(
+        self,
+        ids: None,
+        external_ids: str,
+    ) -> SimulatorRoutineRevision | None: ...
+
+    @overload
+    def retrieve(
+        self,
+        ids: Sequence[int] | None = None,
+        external_ids: SequenceNotStr[str] | None = None,
+    ) -> SimulatorRoutineRevisionList | None: ...
+
+    def retrieve(
+        self,
+        ids: int | Sequence[int] | None = None,
+        external_ids: str | SequenceNotStr[str] | None = None,
+    ) -> SimulatorRoutineRevision | SimulatorRoutineRevisionList | None:
+        """`Retrieve simulator routine revisions <https://developer.cognite.com/api#tag/Simulator-Routines/operation/retrieve_simulator_routine_revisions_simulators_routines_revisions_byids_post>`_
+
+        Retrieve simulator routine revisions by ID or External Id.
+
+        Args:
+            ids (int | Sequence[int] | None): Simulator routine revision ID or list of IDs
+            external_ids (str | SequenceNotStr[str] | None): Simulator routine revision External ID or list of external IDs
+
+        Returns:
+            SimulatorRoutineRevision | SimulatorRoutineRevisionList | None: Requested simulator routine revision
+
+        Examples:
+
+            Get simulator routine revision by id:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.routines.revisions.retrieve(ids=123)
+
+            Get simulator routine revision by external id:
+                >>> res = client.simulators.routines.revisions.retrieve(external_ids="routine_v1")
+        """
+        self._warning.warn()
+        identifiers = IdentifierSequence.load(ids=ids, external_ids=external_ids)
+        return self._retrieve_multiple(
+            resource_cls=SimulatorRoutineRevision,
+            list_cls=SimulatorRoutineRevisionList,
+            identifiers=identifiers,
+            resource_path="/simulators/routines/revisions",
+        )
+
+    @overload
+    def create(self, routine_revision: Sequence[SimulatorRoutineRevisionWrite]) -> SimulatorRoutineRevisionList: ...
```

Comment:
Should these call the create() on line 190? Do we want placeholders like this in the ready code? (also see the one below)

Reply:
just following the sdk pattern here, maybe @cognitedata/python-sdk-maintainers have thoughs?